src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -309,12 +312,21 @@ private static GeneratedSyntax GenerateForAssemblies(List<Assembly> assemblies,
                 var assembly = assemblies[i];
                 foreach (var attribute in assembly.GetCustomAttributes<ConsiderForCodeGenerationAttribute>())
                 {
-                    ConsiderType(attribute.Type, runtime, targetAssembly, includedTypes, attribute.GenerateSerializer);
+                    ConsiderType(attribute.Type, runtime, targetAssembly, includedTypes, true);
+                    if (attribute.ThrowOnFailure && !SerializerGenerationManager.IsTypeRecorded(attribute.Type))
+                    {
+                        throw new CodeGenerationException(
+                            $"Found {attribute.GetType().Name} for type {attribute.Type.GetParseableName()}, but code"
+                            + " could not be generated. Ensure that the type is accessible.");
+                    }
                 }
 
+                KnownAssemblyAttribute knownAssemblyAttribute;
+                var treatAsSerializable = knownAssemblyAttributes.TryGetValue(assembly, out knownAssemblyAttribute)
```

Comment:
I'd rename this variable (and the parameter in ConsiderType) to something like forceCreateSerializer... because a `false` value doesn't mean that no serializer will be created.